### ISSUE TITLE
Support for Policy Assignment metadata

### ISSUE THREAD

**Issue by tyconsulting** [terraform-providers/terraform-provider-azurerm#7724]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Add support for specifying meta data for Azure Policy Assignments. Same as Azure Policy Definitions, metadata is also supported by Policy Assignment in ARM. Currently, the azure_policy_assignment in azurerm provider (v2.18.0) does not support metadata as an input.

### New or Affected Resource(s)

* azurerm_policy_assignment

### Potential Terraform Configuration

```hcl
resource "azurerm_policy_assignment" "this" {
    name         = substr(replace(uuidv5("url", "${var.policy_definition_id}-at-${var.scope}"), "-", ""), 0, 24)
    display_name = var.display_name
    description  = var.description
    metadata     = <<METADATA
    {
        "category": "General"
    }
METADATA
    policy_definition_id  = var.policy_definition_id
    parameters   = var.parameters
    scope        = var.scope
    dynamic identity {
        for_each = var.identity_type == "" ? [] : [1]
        content {
            type = var.identity_type
        }
    }
    location     = var.location
    not_scopes     = var.not_scopes
    enforcement_mode = var.enforcement_mode

}
```

### References

* https://docs.microsoft.com/en-us/azure/templates/microsoft.authorization/2019-06-01/policyassignments

**Comment by tyconsulting** [comment]
PR Created: https://github.com/terraform-providers/terraform-provider-azurerm/pull/7725/, however, it doesn't look like I can link this issue to the PR?

**Comment by ArcturusZhang** [comment]
Hi @tyconsulting thanks for the issue and the PR!

I am reviewing the PR, and if you want to link the PR to this issue, you could try edit your PR description by adding one line of `Fixes #7724` or `Resolves #7724` that starts with a github keyword
